websites/M/MangaHere/presence.ts

```diff
@@ -0,0 +1,140 @@
+    else if (pathname.endsWith("/completed/")) {
+      const url = pathname,
+       splitUrl = url.split('/');
+      if (splitUrl[1] == "completed") {
+        data.details = "Browsing completed manga";
+      } else {
+        data.details = "Browsing completed " + splitUrl[1] + " manga";
+      }
+      data.startTimestamp = browsingStamp;
+    }
+    //ganre/on_going/
+    else if (pathname.endsWith("/on_going/")) {
+      const url = pathname,
+       splitUrl = url.split('/');
+      if (splitUrl[1] == "on_going") {
+        data.details = "Browsing ongoing manga";
+      } else {
+        data.details = "Browsing ongoing " + splitUrl[1] + " manga";
+      }
+      data.startTimestamp = browsingStamp;
+    }
+    //Manga Viewing
+    else if (pathname.startsWith("/manga") && pathname.endsWith("/")) {
+      const title = document.querySelector(".detail-info-right-title-font").textContent,
+       link = window.location.href;
+      data.details = "Viewing manga:";
+      data.state = title;
+      data.startTimestamp = browsingStamp;
+      data.buttons = [{ label: "Read", url: link }];
+      data.smallImageKey = "viewing";
+      data.smallImageText = "Viewing";
+    }
+    //Manga Reading
+    else if (pathname.startsWith("/manga") && pathname.endsWith(".html")) {
+      const title = document.querySelector(".reader-header-title-1").textContent,
+       chapter = document.querySelector(".reader-header-title-2").textContent,
+      //setting up page progress
+       current = document.querySelector('.pager-list-left span');
+      if(current == null){
+        data.state = chapter;
+      }else{
+         const len = current.children.length,
+         totalPages = current.children[len - 2].textContent,
+         readingPage = document.querySelector(".pager-list-left > span > .active").textContent,
+         progress = readingPage + "/" + totalPages;
+        data.state = chapter + " page " + progress;
+      }
+      data.details = title;
+      data.startTimestamp = browsingStamp;
+      data.smallImageKey = "reading";
+      data.smallImageText = "Reading";
+    }
+    //Searching
+    else if (pathname.startsWith("/search")) {
+      const queryString = window.location.search,
+       urlParams = new URLSearchParams(queryString),
+       search = urlParams.get('title');
+       let searchName = "";
+      if (search == "") {
+        searchName = urlParams.get('name');
+      } else {
+        searchName = urlParams.get('title');
+      }
+      data.details = "Searching:";
+      data.state = searchName;
+      data.startTimestamp = browsingStamp;
+      data.smallImageKey = "searching";
+      data.smallImageText = "Searching";
+    }
+    //Browisng ganre
+    for(let i = 0; i<ganres.length; i++){
+      if(pathname.substring(1, pathname.length-1) == ganres[i]){
+        data.details = "Browsing:";
+        data.state = ganres[i].replace('-',' ') +" manga";
+      data.startTimestamp = browsingStamp;
+      }
+    }
```

Comment:
maybe you could use ``.map`` or ``.forEach`` instead?

Reply:
Sure, I went with .forEach. Thank You